pkg/config/validation.go

```diff
@@ -123,6 +123,10 @@ func validateWaitForPodsReady(c *configapi.Configuration) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(waitForPodsReadyPath.Child("timeout"),
 			c.WaitForPodsReady.Timeout, apimachineryvalidation.IsNegativeErrorMsg))
 	}
+	if c.WaitForPodsReady.RecoveryTimeout != nil && c.WaitForPodsReady.RecoveryTimeout.Duration < 0 {
```

Comment:
Does this mean that we can specify 0 in the `recoveryTimeout`? 
In that case, what if recoveryTimeout==0?

Reply:
Yes, in case of `recoveryTimeout == 0` workload is evicted as soon as one of its pod fails

Reply:
In that case, what are the differences between `recoveryTimeout: null` and `recoveryTimeout: 0`?

Reply:
If the `recoveryTimeout == nil` then workload will never be evicted if one of its pods fails

Reply:
That sounds reasonable.
Thank you for clarifying.

/lgtm
/approve